source/extensions/common/wasm/wasm.cc

```diff
@@ -640,6 +762,30 @@ bool Context::setSharedData(absl::string_view key, absl::string_view value, uint
   return global_shared_data.set(wasm_->id(), key, value, cas);
 }
 
+// Shared Queue
+
+uint32_t Context::registerSharedQueue(absl::string_view queue_name) {
+  auto root_context_id = id(); // Assume that Context is a root context.
+  if (root_context_id_ != 0) { // Root contexts have root_context_id_ == 0.
```

Comment:
```
if (root_context_id_ == 0) {
    // Context is a root context.
    root_context_id = id();
} else {
    // Convert context from stream context to root context.
    root_context_id = root_context_id_;
}
```
would avoid extra assignment, and is IMHO slightly more readable.

Reply:
The compiler will eliminate the extra assignment.  What about ? :

Reply:
Done.